src/formatDate.js

```diff
@@ -38,19 +38,88 @@
  *
  * formatDate(
  *   '97/02/18',
- *   ['YY', 'MM', 'DD', '/'],
- *   ['DD', 'MM', 'YYYY', '.'],
- * ) // '18.02.1997'
+ *   fromFormat['YY', 'MM', 'DD', '/'],
+ *   toFormat['DD', 'MM', 'YYYY', '.'],
+ * ) // return '18.02.1997'
  *
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
- * @returns {string}
+ * @returns {string[]}
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (fromFormat[0] === toFormat[0] && fromFormat[2] === toFormat[2]) {
+    return date.split(`${fromFormat[3]}`).join(`${toFormat[3]}`);
+  }
+
+  if (fromFormat[0] === toFormat[2] && fromFormat[2] === toFormat[0]) {
+    return date
+      .split(`${fromFormat[3]}`)
+      .reverse()
+      .join(`${toFormat[3]}`);
+  }
+
+  const splitDate = date.split(`${fromFormat[3]}`);
+
+  if (fromFormat[2] !== toFormat[2] && fromFormat[2] === 'YYYY') {
```

Comment:
You make same check in first `if` block after that proramm ending with `return` statement. You don't need a checks like `fromFormat === toFormat`. Delete this.